src/librustc_data_structures/smallvec.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! A vector type that allows storing small values directly on the stack.
```

Comment:
To some extent, yes; storing huge values on stack is also a no-go, but the vector itself has nothing to do with that. Can, and will, update if given the go-ahead. Probably warrants including a longer comment too (addressing some of your other concerns, perhaps).